### PR TITLE
fix kubernetes example, set the ballast size in the config file

### LINKED ISSUE #2213: Outdated Kubernetes instructions
### Summary

I tried deploying the Kubernetes version of the collector ([this one](https://github.com/signalfx/splunk-otel-collector/blob/8578955e71e77decc56720d0e1b71d699b82948b/examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml)) and it gives me the following error:
```
2022/11/02 11:31:48 settings.go:269: Set config to [/conf/otel-collector-config.yaml]
2022/11/02 11:31:48 settings.go:333: Set ballast to 168 MiB
2022/11/02 11:31:48 settings.go:349: Set memory limit to 460 MiB
Error: unknown flag: --mem-ballast-size-mib
2022/11/02 11:31:48 main.go:99: application run finished with error: unknown flag: --mem-ballast-size-mib
```

What do we have to do with [that flag](https://github.com/signalfx/splunk-otel-collector/blob/8578955e71e77decc56720d0e1b71d699b82948b/examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml#L205)?

Related PRs: https://github.com/signalfx/splunk-otel-collector/pull/567

### How to reproduce

```
❯ docker run --rm -e SPLUNK_REALM=us1 -e SPLUNK_ACCESS_TOKEN=test quay.io/signalfx/splunk-otel-collector:0.62.0 /otelcol --mem-ballast-size-mib=168
2022/11/02 11:33:20 settings.go:286: Set config to /etc/otel/collector/gateway_config.yaml
2022/11/02 11:33:20 settings.go:333: Set ballast to 168 MiB
2022/11/02 11:33:20 settings.go:349: Set memory limit to 460 MiB
Error: unknown flag: --mem-ballast-size-mib
2022/11/02 11:33:20 main.go:99: application run finished with error: unknown flag: --mem-ballast-size-mib
```

### PR DESCRIPTION
**Pull request by atoulme**
Fixes #2213

### CODE REVIEW

**Comment by rmfitzpatrick** [comment]
I believe the issue #2213 is reporting is that we aren't stripping the cli option and argument and it's been unrecognized in the collector service since https://github.com/open-telemetry/opentelemetry-collector/pull/3626. This would require enhancing `settings.removeFlag` in addition to removing the option from the service args.

**Comment by atoulme** [comment]
OK, I think we can handle this separately, I'll file a ticket - at least this fixes the example. Does that sound good?

File: examples/kubernetes-yaml/splunk-otel-collector-agent.yaml [hunk 1]
```diff
@@ -18,6 +18,12 @@ data:
           # TODO: Ensure this is set properly
           endpoint: "https://api.${SPLUNK_REALM}.signalfx.com"
       zpages:
+      memory_ballast:
+        # In general, the ballast should be set to 1/3 of the collector's memory, the limit
+        # should be 90% of the collector's memory.
+        # The simplest way to specify the ballast size is set the value of SPLUNK_BALLAST_SIZE_MIB env variable.
+        # TODO: Ensure this is set properly
```
[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply examples/kubernetes-yaml/splunk-otel-collector-agent.yaml:25]: oh sry, I guess that todo is for the user

**Review by rmfitzpatrick — APPROVED** [review]

File: examples/kubernetes-yaml/splunk-otel-collector-agent.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (atoulme) [reply examples/kubernetes-yaml/splunk-otel-collector-agent.yaml:25]: I think this example is leaving a lot of work to the user :|

**Pull request merged**